modules/data/event-hub/main.bicep

```diff
@@ -0,0 +1,245 @@
+Below paramters you can pass while the creating Azure Event Hub
+messageRetentionInDays: (Optional) int, Number of days to retain the events for this Event Hub, value should be 1 to 7 days. Default to 1.
+partitionCount: (Optional) int, Number of partitions created for the Event Hub. Default to 2
+eventHubNamespaceName: (Optional) string, Name of the Azure Event Hub Namespace
+status: (Optional) string, Enumerates the possible values for the status of the Event Hub. 'Active','Creating','Deleting','Disabled','ReceiveDisabled','Renaming','Restoring','SendDisabled','Unknown'. Default to 'Active'
+''')
+param eventHubs object = {}
+
+@description('Optional. Authorization Rules for the Event Hub Namespace.')
+param namespaceAuthorizationRules object = {}
+
+@description('Optional. Authorization Rules for the Event Hub .')
+param eventHubAuthorizationRules object = {}
+
+@description('Optional. consumer groups for the Event Hub .')
+param consumerGroups object = {}
+
+@description('Optional. The disaster recovery config for the namespace.')
+param disasterRecoveryConfigs object = {}
+
+@description('Optional. The Diagnostics Settings config for the namespace.')
+param diagnosticSettings object = {}
+
+var varEventHubNamespaces = [for eventHubnamespace in items(eventHubNamespaces): {
+  eventHubNamespaceName: eventHubnamespace.key
+  sku: contains(eventHubnamespace.value, 'sku')? eventHubnamespace.value.sku : 'Standard'
+  capacity: contains(eventHubnamespace.value, 'capacity') ? eventHubnamespace.value.capacity: 1
+  zoneRedundant: contains(eventHubnamespace.value, 'zoneRedundant') ? eventHubnamespace.value.zoneRedundant : false
+  isAutoInflateEnabled: contains(eventHubnamespace.value, 'isAutoInflateEnabled') ? eventHubnamespace.value.isAutoInflateEnabled: false
+  maximumThroughputUnits: contains(eventHubnamespace.value, 'maximumThroughputUnits') ? eventHubnamespace.value.maximumThroughputUnits: 0
+  disableLocalAuth: contains(eventHubnamespace.value, 'disableLocalAuth') ? eventHubnamespace.value.disableLocalAuth: false
+  kafkaEnabled: contains(eventHubnamespace.value, 'kafkaEnabled') ? eventHubnamespace.value.kafkaEnabled: true
+}]
+
+var varNamespaceAuthorizationRules =  [for namespaceAuthorizationRule in items(namespaceAuthorizationRules): {
+  eventHubNamespaceAuthorizationRuleName: namespaceAuthorizationRule.key
+  rights: contains(namespaceAuthorizationRule.value, 'rights') ? namespaceAuthorizationRule.value.rights : []
+  eventHubNamespaceName: namespaceAuthorizationRule.value.eventHubNamespaceName
+}]
+
+var varDisasterRecoveryConfigs =  [for disasterRecoveryConfig in items(disasterRecoveryConfigs): {
+  disasterRecoveryConfigname: disasterRecoveryConfig.key
+  partnerNamespaceId: contains(disasterRecoveryConfig.value, 'partnerNamespaceId') ? disasterRecoveryConfig.value.partnerNamespaceId : ''
+  eventHubNamespaceName: disasterRecoveryConfig.value.eventHubNamespaceName
+}]
+
+var varEventHubs =  [for eventHub in items(eventHubs): {
```

Comment:
Great work here! Having this kind of stuff in the template brings a ton of value.